master/buildbot/buildslave/ec2.py

```diff
@@ -120,15 +120,14 @@ def __init__(self, name, password, instance_type, ami=None,
             if aws_id_file_path is None:
                 home = os.environ['HOME']
                 aws_id_file_path = os.path.join(home, '.ec2', 'aws_id')
-            if not os.path.exists(aws_id_file_path):
-                raise ValueError(
-                    "Please supply your AWS access key identifier and secret "
-                    "access key identifier either when instantiating this %s "
-                    "or in the %s file (on two lines).\n" %
-                    (self.__class__.__name__, aws_id_file_path))
-            with open(aws_id_file_path, 'r') as aws_file:
-                identifier = aws_file.readline().strip()
-                secret_identifier = aws_file.readline().strip()
+                if not os.path.exists(aws_id_file_path):
+                    aws_id_file_path = None
+            if aws_id_file_path:
+                log.msg('WARNING: EC2LatentBuildSlave is using deprecated '
```

Comment:
@sa2ajj I believe this already does what you want, but it might be clearer if I wasn't using `aws_id_file_path` as temporary storage between 120 and 124. I've added a new variable, the actual logic hasn't changed. The warning will trigger if `aws_id_file_path` is set or the default file exists.